shell.nix

```diff
@@ -0,0 +1,12 @@
+{ pkgs ? import <nixpkgs> { } }:
```

Comment:
do we want to add this this file in this PR ? I think I added it because I use nixos but we should probably not have it ? (mistake from my PR I need to fix.)

Reply:
I also use nixos :smile: and would find it convenient to have it in the project. It should no cause any harm for people not using nix.